interbotix_ros_xseries/interbotix_xs_sdk/src/xs_sdk_obj.cpp

```diff
@@ -532,10 +549,18 @@ bool InterbotixRobotXS::robot_get_motor_configs(void)
     std::string gripper_name = gripper_itr->first.as<std::string>();
     YAML::Node single_gripper = all_grippers[gripper_name];
     Gripper gripper;
+    gripper.type = single_gripper["type"].as<std::string>("swing_arm");
+    if (gripper.type != "swing_arm" && gripper.type != "rack_and_pinion")
+    {
+      ROS_ERROR("[xs_sdk] Invalid gripper type. Please check the config.yaml file");
+      return false;
+    }
     gripper.horn_radius = single_gripper["horn_radius"].as<float>(0.014);
+    gripper.pitch_radius = single_gripper["pitch_radius"].as<float>(0.0127);
     gripper.arm_length = single_gripper["arm_length"].as<float>(0.024);
     gripper.left_finger = single_gripper["left_finger"].as<std::string>("left_finger");
     gripper.right_finger = single_gripper["right_finger"].as<std::string>("right_finger");
+    gripper.calibrate = single_gripper["calibrate"].as<bool>(true);
```

Comment:
Don't calibrate by default
```suggestion
    gripper.calibrate = single_gripper["calibrate"].as<bool>(false);
```